### ISSUE TITLE
AWS Chime SDK Doesn't work with React Native

### ISSUE THREAD

**Issue by robertjcolley** [aws/amazon-chime-sdk-js#1856]
### What happened and what did you expect to happen?

React Native supports websockets and can handle it. I expected messaging to work, but I got an error

> Error: null is not an object (evaluating 'this.version().split()')

I ended up tracing it to the detect-browser React Native implementation here:

https://github.com/DamonOehlman/detect-browser/blob/546e6f1348375d8a486f21da07b20717267f6c49/src/index.ts#L59-L65

which is referenced in 

https://github.com/aws/amazon-chime-sdk-js/blob/8af993e501fbf0c18b0b567e5e5a20114f08abd2/src/browserbehavior/DefaultBrowserBehavior.ts

The solution is:

1. detect-browser needs to update their implementation of React Native. I've made a PR to their project ([HERE](https://github.com/DamonOehlman/detect-browser/pull/172))
2. aws-chime-sdk-js needs to update detect-browser when this PR is merged in and detect-browser is updated

I've been able to successfully create a patch which I can use [patch-package](https://github.com/ds300/patch-package) to run in a post install script for the time being (see attached) and confirmed that this patch did correctly fix the issue and I was successfully able to use the SDK. I'm creating this issue here to track this issue until it is resolved, and will update this as I fix issues in detect-browser.

I had to rename the ".patch" file to ".patch.txt" to upload to GitHub:

[detect-browser+5.3.0.patch.txt](https://github.com/aws/amazon-chime-sdk-js/files/7671567/detect-browser%2B5.3.0.patch.txt)

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

Try creating a DefaultMessagingSession in React Native

### Amazon Chime SDK for JavaScript version

latest

### What browsers are you seeing the problem on?

React Native

### Browser version

0.65.1

### Meeting and Attendee ID Information.

N/A

### Browser console logs

N/A - see top section

**Comment by simmkyu** [comment]
Thanks for creating an issue. In addition to WebSocket, Chime SDK for JavaScript uses WebRTC, MediaDevices, Web Audio, and other browser APIs that may not or partially be supported in React Native, e.g., [The WebRTC module for React Native](https://github.com/react-native-webrtc/react-native-webrtc).
 
For React Native use cases, we have a demo application using Chime SDK for iOS and Android. Could you check out the following demo?
https://github.com/aws-samples/amazon-chime-react-native-demo

**Comment by NS-BOBBY-C** [comment]
I have checked that out and have no issues with the meeting functionality provided by those SDKs! They work great 😄.

This issue came about after implementing the messaging (which is implemented in web sockets - something that react native supports). The patch file I provided was enough for this to be completely fixed.

Not an issue with this project, I just need something fixed in the detect-browser package. I made the issue here for when they get back to me with a fix, so I can recommend a dependency upgrade!
